site/utils/apiUtils/createDisruptionValidations.ts

```diff
@@ -0,0 +1,278 @@
+        ) {
+            errors.push(error);
+        }
+    }
+};
+
+export const validateDateTime = (
+    date: Date | null,
+    time: string,
+    errors: ErrorInfo[],
+    errorId: string,
+    dateType: string,
+): boolean => {
+    let isValid = false;
+
+    const timeCheck = requireFieldCheck(time);
+    if (!date) {
+        errors.push({
+            id: errorId,
+            errorMessage: `No ${dateType} date selected. Please select a valid ${dateType} date`,
+        });
+    }
+
+    if (!timeCheck) {
+        errors.push({
+            id: errorId,
+            errorMessage: `No ${dateType} time entered. Please select a valid ${dateType} time`,
+        });
+    }
+
+    if (date && timeCheck) {
+        const jsDate = dayjs(date, CD_DATE_FORMAT, "en-gb", true);
+        const isValidTime = validateTime(time, dateType, errors, errorId);
+
+        if (!jsDate.isValid()) {
+            errors.push({
+                id: errorId,
+                errorMessage: `Invalid ${dateType} Date value submitted. Please select a valid ${dateType} date`,
+            });
+        } else {
+            // Validate that the start time input is valid and of time format
+            const isFutureDate = jsDate.hour(+time.slice(0, 2)).minute(+time.slice(2, 4)).isAfter(dayjs());
+
+            if (isValidTime && !isFutureDate) {
+                errors.push({
+                    id: errorId,
+                    errorMessage: `The ${dateType} date and time should be past the current date and time`,
+                });
+            } else {
+                isValid = isValidTime;
+            }
+        }
+    }
+
+    return isValid;
+};
+
+export const validateTime = (time: string, dateType: string, errors: ErrorInfo[], errorId: string): boolean => {
+    let isValid = false;
+
+    if (time.length != 4) {
+        errors.push({
+            id: errorId,
+            errorMessage: `Value for ${dateType} time is of invalid format. Please enter the time in 24hr format. For example 0900 is 9am, 1730 is 5:30pm`,
+        });
+    } else if (isNaN(+time)) {
+        errors.push({
+            id: errorId,
+            errorMessage: `Value for ${dateType} time is not a Number. Please enter the time in 24hr format. For example 0900 is 9am, 1730 is 5:30pm`,
+        });
+    } else {
+        isValid = true;
+        if (!(+time.slice(0, 2) < 24)) {
+            errors.push({
+                id: errorId,
+                errorMessage: `Invalid hour value entered in ${dateType} time. The hour value should be between 0-23`,
+            });
+            isValid = false;
+        }
+
+        if (!(+time.slice(2, 4) < 60)) {
+            errors.push({
+                id: errorId,
+                errorMessage: `Invalid minutes value entered in ${dateType} time. The minutes values should be between 0-59`,
+            });
+
+            isValid = false;
+        }
```

Comment:
Again, I think we can combine these errors to avoid the user having multiple things to read. 

```
if (!(+time.slice(0, 2) < 24) || !(+time.slice(2, 4) < 60) ) {

errors.push({
                id: errorId,
                errorMessage: {error},
            });

}
```

Reply:
Ideally we dont want one input to be creating multiple errors for the user.